### PR TITLE
Do not add a new line in multilevel chained package clause

### PR DESCRIPTION
**Pull request by iRevive**
This PR fixes https://github.com/scalameta/scalafmt/issues/1593

### CODE REVIEW

File: scalafmt-tests/src/test/resources/newlines/alwaysBeforeTopLevelStatements.source [hunk 1]
```diff
@@ -36,3 +36,16 @@ package b {
...
+package f
+import g.h.i.j
+package k {}
+>>>
+package a.b.c
+package d.e
+package f
+
+import g.h.i.j
+package k {}
```
Comment (kitbellew) [inline scalafmt-tests/src/test/resources/newlines/alwaysBeforeTopLevelStatements.source]: should there be an empty line here as well? perhaps it was this way before.

File: scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala [hunk 2]
```diff
@@ -259,24 +259,35 @@ class FormatWriter(formatOps: FormatOps) {
...
             .collect { case term: Term.Name => term.parent }
             .flatten
-            .collect { case pkg: Pkg => pkg.stats.headOption }
+            .collect {
+              // package a
+              case pkg: Pkg =>
+                pkg.stats.headOption
+
+              // package a.b.c
+              case select: Term.Select =>
```
Comment (kitbellew) [inline scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala]: would this pattern match `package a.b.c {` as well? there might an `Apply` there somewhere.

**Comment by kitbellew** [comment]
@iRevive thank you for the fix, and for adding the test.

@olafurpg the commit is good to go.

File: scalafmt-tests/src/test/resources/newlines/alwaysBeforeTopLevelStatements.source (hunk 1 above)
Comment (iRevive) [reply scalafmt-tests/src/test/resources/newlines/alwaysBeforeTopLevelStatements.source]: Currently, scalafmt does not change formatting between import and class and keeps an empty line (if it present) as is.

**Comment by poslegm** [comment]
Thank you!

**Pull request merged**